pkg/controller/daemon/daemoncontroller.go

```diff
@@ -753,6 +753,32 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 	newPod.Namespace = ds.Namespace
 	newPod.Spec.NodeName = node.Name
 
+	// DaemonSet pods shouldn't be deleted by NodeController in case of node problems.
+	// Add infinite toleration for taint notReady:NoExecute here
+	// to survive taint-based eviction enforced by NodeController
+	// when node turns not ready.
+	_, err = v1.AddOrUpdateTolerationInPod(newPod, &v1.Toleration{
```

Comment:
Well - you can't do this now, so this is not a breaking change. We can figure a way for you to do it later.